src/components/InstantSearch.js

```diff
@@ -0,0 +1,91 @@
+import instantsearch from 'instantsearch.js/es/';
+import { createInstantSearchComponent } from '../util/createInstantSearchComponent';
+import { warn } from '../util/warn';
+
+const oldApi = () =>
```

Comment:
Instead of creating an implicit function for that, we can store the message in a variable and warn it everywhere we want.